### PR TITLE
add mech kinetic accelerator

### PR DESCRIPTION
**Pull request by pyronie**
# Document the changes in your pull request
adds a mech Kinetic accelerator and adds it to the vendor for 10k points. I tested this and it seems to work fine as far as i could tell. it is locked to the vendor so that roboticists can't powergame with it, and added to all vendors so that golems and freeminers can also use it. also fixed some of the formatting in mining vendor and added an admin only version that has no damage reduction in pressurized environments and fits all mechs.

# Spriting
image_2022-07-15_004533109

# Wiki Documentation
Mech Kinetic accelerator, only attachable to mining mechs and does 100 damage in depressurized environments, 20 in pressurized environments, had a range of 5 and has a 1 second cooldown. costs 10000 points in the mining vendor and is available in all mining vendors. note that you can achieve similar results without a mech in lavaland with dual wielding two kinetic accelerators with a range, damage, and cooldown modkit.

# Changelog
:cl:  
rscadd: mining mech KA, only in mining vendors for 10k points
imageadd: sprite for the mech KA
spellcheck: fixed mining vendors code formatting
/:cl:

### CODE REVIEW

**Comment by pyronie** [comment]
if anyone thinks this is too strong and needs to be made weaker, please remember you can make a pair of kinetic accelerators that are essentially better than this, with two ka's with 2 damage upgrades and 1 cooldown upgrade each, they deal more damage than this at a negligibly slower rate.

**Comment by ShadowDeath6** [comment]
What's the point of this? Just to give ripley mechs a ranged option? I don't really mind, but I thought that the plasma cutter already did fairly solid damage.

**Comment by ghost** [comment]
If i would want to powergame as a roboticist, i would build a Durand. No need to lock it to miners only, because mostly KA is useless on station, since it deals small amounts of damage in low pressure.

**Comment by ghost** [comment]
Also, is the mech KA upgradeable?

**Review by ToasterBiome — CHANGES_REQUESTED** [review]
You messed up the line endings, please fix

**Comment by pyronie** [comment]
> If i would want to powergame as a roboticist, i would build a Durand. No need to lock it to miners only, because mostly KA is useless on station, since it deals small amounts of damage in low pressure.

does 20 dps on station and would be equippable on util mechs

File: code/modules/projectiles/guns/energy/kinetic_accelerator.dm [hunk 1]
```diff
@@ -1,604 +1,630 @@
...
+	var/power = 1
+
+	var/pressure_decrease_active = FALSE
+	var/pressure_decrease = 0.25
+	var/obj/item/gun/energy/kinetic_accelerator/kinetic_gun
+
+/obj/item/projectile/kinetic/mech
+	name = "kinetic force"
+	icon_state = null
+	damage = 100
```
Comment (ToasterBiome) [inline code/modules/projectiles/guns/energy/kinetic_accelerator.dm:192]: Hell no

**Comment by pyronie** [comment]
> You messed up the line endings, please fix

i would but its due to how github processes my upload so idk how to fix

**Review by ToasterBiome — COMMENTED** [review]
same amount of damage as a backstab crusher while being 7 tiles away... holy shit.........

**Comment by ToasterBiome** [comment]
> > You messed up the line endings, please fix
> 
> i would but its due to how github processes my upload so idk how to fix

use github desktop or sourcetree or any other programs to do it properly

File: code/modules/projectiles/guns/energy/kinetic_accelerator.dm (hunk 1 above)
Comment (pyronie) [reply code/modules/projectiles/guns/energy/kinetic_accelerator.dm:192]: thats 5 tiles away, not 7, can be easily be surpassed by dualies set up right, and is on something less mobile than a player

**Review by pyronie — COMMENTED** [review]
reduced range and increased cooldown

**Comment by pyronie** [comment]
pretty sure i can only fix the line end is by closing this pr and starting a new one because of how i made the change

**Comment by pyronie** [comment]
> Also, is the mech KA upgradeable?

no

**Comment by ghost** [comment]
> does 20 dps on station and would be equippable on util mechs

It has short range, long cooldown, and isn't upgradeable. Roboticists have more powerfull weapons in their disposal, so i don't think that locking it to mining vendors and making it cost fucking 10k points is a good idea.

**Comment by ghost** [comment]
Ah stop it isn't a standart KA. But actually 100 damage is a fucking enormous value, are you sure about it?

**Comment by KillerOrcaCora** [comment]
https://youtu.be/GjXVUTPTKdU now I can finally beat mega funna with my ass blast gg from 7 tiles away

**Comment by pyronie** [comment]
> Ah stop it isn't a standart KA. But actually 100 damage is a fucking enormous value, are you sure about it?

specifically 100 damage so that it can oneshot legions. note that this is supposedly a completed KA so it can't be modified. also is locked from roboticists because it does 20 damage on station, which while isnt as much, it isnt a weapon and can be fitted on the utility mechs. it would suddenly be the best weapon on utility mechs by far especially because it actually is usable.

> https://youtu.be/GjXVUTPTKdU now I can finally beat mega funna with my ass blast gg from 7 tiles away

the range and firing speed has been reduced, and it doesnt have 7 range. also note despite it having 5 range in that clip(it now has 4) i still sometimes am too far away to hit the colossus.

File: code/modules/projectiles/guns/energy/kinetic_accelerator.dm (hunk 1 above)
Comment (pyronie) [reply code/modules/projectiles/guns/energy/kinetic_accelerator.dm:192]: was reduced to 4 range and increased cd

**Comment by pyronie** [comment]
me killing colossus with range 4 and increased cd mech ka 
https://youtu.be/8WXXVSv5qtQ

**Comment by pyronie** [comment]
also @TymurShatillo @ToasterBiome a mega KA can exceed the current dps of the mech ka. currently it has 62.5 dps, while a mega ka with 3 cd reductions and 1 damage upgrade has a dps of around 75. this is why i said when you are complaining about this being strong you are really just complaining about ka's being strong.

**Comment by ToasterBiome** [comment]
> also @TymurShatillo @ToasterBiome a mega KA can exceed the current dps of the mech ka. currently it has 62.5 dps, while a mega ka with 3 cd reductions and 1 damage upgrade has a dps of around 75. this is why i said when you are complaining about this being strong you are really just complaining about ka's being strong.

Getting a mega KA involves finding basalt, getting plasma magmite, finding the world anvil, and forging it with a piece of gibtonite

On the other hand, you AFK request a ripley from robotics and then spend 10K on it and you get a free colossus kill

**Comment by pyronie** [comment]
the first one is easier and also is not on a mech

**Comment by ghost** [comment]
Basicaly if the mech KA has the same range and cooldown as the normall one it is less strong then just going 2 KA with two damage upgrades plus range, because it will be 120 damage from one attack cycle, so maybe it is OK

**Comment by KillerOrcaCora** [comment]
Here is a super fun idea I came up with while eating garbage. Just make it do no damage to megas!!!

**Comment by ghost** [comment]
bad

**Comment by JamieD1** [comment]
Looks way too easy from that video, the monsters are balanced around people not mechs

**Comment by githubuser4141** [comment]
> What's the point of this? Just to give ripley mechs a ranged option? I don't really mind, but I thought that the plasma cutter already did fairly solid damage.

It does [10 damage](https://github.com/yogstation13/Yogstation/blob/707c6a724d5e03ad7fd3d0b94506f1ded916b506/code/modules/projectiles/projectile/special/plasma.dm) in pressurized or unpressurized areas
AusStation had a mecha KA that started off as 60 damage and could be upgraded, maybe port that one?
Utility mechs do have weapons like the clamp, which does 20 brute, 5 suffo through ALL armor.
But uhhh don't KAs deal explosive damage? This'll put a lot of people who don't have BOMB armor in crit in two shots, which is most of security, engi, science, etc, and it'll take out fellow mechs (Phazons, durands, gygaxes) in 2-3 shots because non-Ripley mechs have low BOMB armor.

**Review by KillerOrcaCora — APPROVED** [review]
Sprites fine good luck on that balance though

**Comment by JamieD1** [comment]
Will bring up with the new balance council

**Review by redmoogle — CHANGES_REQUESTED** [review]
Add an exemption for the PKA in the combat design at `code/game/mecha/mech_fabricator.dm:505` since it falls under the /weapon subtype

**Comment by JamieD1** [comment]
This is being declined because it is basically free megafauna if you know what you are doing, you either need to have a downside and not be just better then everything else or this is being declined, sorry. Reopen it if you change it.

**Pull request closed without merging**